cmake/OpenCVFindLibsGrfmt.cmake

```diff
@@ -36,13 +36,38 @@ if(WITH_JPEG)
   if(NOT JPEG_FOUND)
     ocv_clear_vars(JPEG_LIBRARY JPEG_LIBRARIES JPEG_INCLUDE_DIR)
 
-    set(JPEG_LIBRARY libjpeg)
-    set(JPEG_LIBRARIES ${JPEG_LIBRARY})
-    add_subdirectory("${OpenCV_SOURCE_DIR}/3rdparty/libjpeg")
-    set(JPEG_INCLUDE_DIR "${${JPEG_LIBRARY}_SOURCE_DIR}")
+    if(NOT BUILD_JPEG_TURBO_DISABLE)
+      set(JPEG_LIBRARY libjpeg-turbo)
+      set(JPEG_LIBRARIES ${JPEG_LIBRARY})
+      add_subdirectory("${OpenCV_SOURCE_DIR}/3rdparty/libjpeg-turbo")
+      set(JPEG_INCLUDE_DIR "${${JPEG_LIBRARY}_SOURCE_DIR}/src")
+    else()
+      set(JPEG_LIBRARY libjpeg)
+      set(JPEG_LIBRARIES ${JPEG_LIBRARY})
+      add_subdirectory("${OpenCV_SOURCE_DIR}/3rdparty/libjpeg")
+      set(JPEG_INCLUDE_DIR "${${JPEG_LIBRARY}_SOURCE_DIR}")
+    endif()
   endif()
 
-  ocv_parse_header("${JPEG_INCLUDE_DIR}/jpeglib.h" JPEG_VERSION_LINES JPEG_LIB_VERSION)
+  if(NOT DEFINED JPEG_LIB_VERSION AND EXISTS "${JPEG_INCLUDE_DIR}/jpeglib.h")
+    ocv_parse_header("${JPEG_INCLUDE_DIR}/jpeglib.h" JPEG_VERSION_LINES JPEG_LIB_VERSION)
+  endif()
+  if(NOT DEFINED JPEG_LIB_VERSION AND EXISTS "${JPEG_INCLUDE_DIR}/jconfig.h")  # libjpeg-turbo stores version in a separate file
+    ocv_parse_header("${JPEG_INCLUDE_DIR}/jpeglib.h" JPEG_VERSION_LINES JPEG_LIB_VERSION)
```

Comment:
Why is `jconfig.h` checked for existence but `jpeglib.h` is parsed?

Maybe it is better to make it in a loop, something like:
```.cmake
foreach(jpeg_dir "${JPEG_INCLUDE_DIR};${${JPEG_LIBRARY}_BINARY_DIR}")
foreach(jpeg_hdr "jpeglib.h;jconfig.h;jconfig-64.h;...")
  ocv_parse_header("${jpeg_dir}/${jpeg_hdr}" JPEG_VERSION_LINES JPEG_LIB_VERSION)
  if (JPEG_LIB_VERSION)
    break()
  endif()
endforeach()
endforeach()
```

Reply:
Added macro to avoid copy-paste mistakes